src/odemis/acq/align/test/fastem_test.py

```diff
@@ -0,0 +1,178 @@
+        It calibrates the descanner offset."""
+
+        calibrations = [IMAGE_TRANSLATION_PREALIGN]
+
+        # get current descanner offset
+        descan_offset_cur = self.descanner.scanOffset.value
+
+        # Run image translation pre-align
+        f = fastem.align(self.scanner, self.multibeam, self.descanner, self.mppc, self.stage, self.ccd,
+                         self.beamshift, self.det_rotator, calibrations)
+
+        e = f.result(timeout=900)
+
+        self.assertIsNone(e)  # check no exceptions were returned
+
+        # get the calibrated descanner offset
+        descan_offset_calib = self.descanner.scanOffset.value
+
+        # check the calibrated descan offset is different from the previous offset
+        self.assertNotEqual(descan_offset_cur, descan_offset_calib)
+
+    def test_progress(self):
+        """Check if some progress is reported during the optical autofocus calibration."""
+
+        self.updates = 0  # updated in callback on_progress_update
+
+        calibrations = [OPTICAL_AUTOFOCUS, IMAGE_TRANSLATION_PREALIGN]
+        f = fastem.align(self.scanner, self.multibeam, self.descanner, self.mppc, self.stage, self.ccd,
+                         self.beamshift, self.det_rotator, calibrations)
+
+        f.add_update_callback(self.on_progress_update)  # callback executed every time f.set_progress is called
+        f.add_done_callback(self.on_done)  # callback executed when f.set_result is called (via bindFuture)
+
+        e = f.result()
+
+        self.assertIsNone(e)  # check no exceptions were returned
+        self.assertTrue(self.done)
+        # at least one update per calibration plus once at start of calibration, plus once at end of calibration
+        self.assertGreaterEqual(self.updates, 4)
+
+    def test_cancel(self):
+        """Test if it is possible to cancel the optical autofocus calibration."""
+
+        # FIXME no subfuture available yet, which are cancelable
+        #  when subfutures are implemented, add a check in this test case that the subfuture was also cancelled
+
+        self.end = None  # updated in callback on_progress_update
+        self.updates = 0  # updated in callback on_progress_update
+        self.done = False  # updated in callback on_done
+
+        calibrations = [OPTICAL_AUTOFOCUS]
+        f = fastem.align(self.scanner, self.multibeam, self.descanner, self.mppc, self.stage, self.ccd,
+                         self.beamshift, self.det_rotator, calibrations)
+
+        f.add_update_callback(self.on_progress_update)  # callback executed every time f.set_progress is called
+        f.add_done_callback(self.on_done)  # callback executed when f.set_result is called (via bindFuture)
+
+        time.sleep(1)  # make sure it's started
+        self.assertTrue(f.running())
+        f.cancel()
+
+        self.assertRaises(CancelledError, f.result, 1)  # add timeout = 1s in case cancellation error was not raised
```

Comment:
Please use `with self.assertRaises()`.

Also, I don't get why you're using such a short timeout, especially right now the code actually doesn't cancel the current calibration, so you'd have to wait for the autofocus to end. It doesn't always take < 1s.